src/circuit.rs

```diff
@@ -92,6 +92,63 @@ pub struct Cell {
     column: Column<Any>,
 }
 
+/// An allocated cell.
+#[derive(Clone, Copy, Debug)]
+pub struct AllocatedCell<F: FieldExt, T: Copy + Into<F>> {
+    value: Option<T>,
+    cell: Cell,
+    _marker: PhantomData<F>,
+}
+
+impl<F: FieldExt, T: Copy + Into<F>> AllocatedCell<F, T> {
+    /// Instantiate a new AllocatedCell.
+    pub(crate) fn new(value: Option<T>, cell: Cell) -> Self {
+        Self {
+            value,
+            cell,
+            _marker: PhantomData,
+        }
+    }
+
+    /// Returns the value of the AllocatedCell.
+    pub fn value(&self) -> Option<T> {
+        self.value
+    }
+
+    /// Returns the value of the AllocatedCell as a field element.
+    pub fn value_field(&self) -> Option<F> {
+        self.value.map(|v| v.into())
+    }
+
+    /// Returns the cell.
+    pub fn cell(&self) -> Cell {
+        self.cell
+    }
+
+    /// Copies the value to a given advice cell and constrains them to be equal.
```

Comment:
This isn't constraining them to be equal though, as far as I can tell? It's just assigning.